src/services/intentService.ts

```diff
@@ -16,15 +17,26 @@ export async function publishIntent(
   userInfo: { userAddress: string; userChainType: ChainType },
   quoteHashes: string[]
 ): Promise<PublishIntentResult> {
-  // todo: retry on network error
-  const result = await solverRelayClient.publishIntent({
-    signed_data: prepareSwapSignedData(signatureData, userInfo),
-    quote_hashes: quoteHashes,
-  })
-
-  if (result.status === "OK") return { tag: "ok", value: result.intent_hash }
-
-  return { tag: "err", value: { reason: result.status } }
+  return retry<PublishIntentResult>(
+    async () => {
+      const result = await solverRelayClient.publishIntent({
+        signed_data: prepareSwapSignedData(signatureData, userInfo),
+        quote_hashes: quoteHashes,
+      })
+      if (result.status === "OK")
+        return { tag: "ok", value: result.intent_hash }
+      if (result.status === "FAILED" && result.reason === "already processed")
+        return { tag: "ok", value: result.intent_hash }
+      return { tag: "err", value: { reason: result.status } }
+    },
+    {
+      delay: 1000,
+      factor: 1.5,
+      maxAttempts: 7,
+      jitter: true,
+      minDelay: 1000,
+    }
+  )
```

Comment:
The order should be slightly different. Handling should go after retry.
```js
const res = await retry(() => myFn())
doSomething(res)
```

Reply:
But it worked hm